Materialize site

subnomo opened the pull request:
<!--
  This is a guideline for what a PR should look like.
  Feel free to modify it to fit your specific needs.

  Please list the issue this fixes in the PR
-->

Fixes #33 

This PR still needs work to improve usability, but I believe all that's left are "nice-to-haves," basic functionality should be there. If something is actually broken, please let me know.
## Changes in this PR.

<!--
  Please include a list of all things this PR will accomplish
  Use the checkbox syntax to show what is done and what is yet to be completed.
  This gives context to the diff.
-->
- [X] Materialize existing pages
## Testing this PR.

<!--
  Please include a list of explicit instructions for testing this PR.
  If the instructions are 'run `make test`' say that,
  If they are more complicated be thorough.
-->
1. Run `docker-compose build`
2. Start the container with `docker-compose up`
3. Visit http://localhost:5000, browse the site
### Expected Output.

<!--
  Please insert either a description of what should happen when testing
  your PR or a code-block with terminal output.
-->

You should see the materialized pages. This is an example from the `/times` page:
![times page](https://cloud.githubusercontent.com/assets/1390653/19662613/215f8c2e-99ed-11e6-9440-29b6e7442443.png)

@osuosl/devs

subnomo committed:
Cleanup scripts

subnomo committed:
Reorganize files and use Flask-Assets

subnomo committed:
Ignore compiled static files

subnomo committed:
Add sidebar, reorganize file structure

subnomo committed:
Cleanup css, move sidebar to layout template

subnomo committed:
Reorganize sidebar links

subnomo committed:
Adding materialize date picker

subnomo committed:
Fix datepicker

subnomo committed:
Fix view_times form

subnomo committed:
Fix & materialize login page

subnomo committed:
Use lock icon

subnomo committed:
WIP: edit_time form broken...

subnomo committed:
Fix edit_time and delete_time views

subnomo committed:
Fix create_time page

subnomo committed:
Remove edit and delete links from sidebar

subnomo committed:
Add simple welcome message to index page

subnomo committed:
Add permissions when rendering template

subnomo committed:
Fix tests

subnomo committed:
Materialize buttons

subnomo committed:
Add is_logged_in check to login page

Just in case the user *is* logged in and wants to see the sidebar items.

subnomo committed:
Fix checkboxes on create_user page

subnomo committed:
Remove upper right link to create_* pages

aaroncohen73 commented on app/templates/create_time.html:
```diff
@@ -6,10 +6,30 @@
+        {% if form.activities.choices %}
+            <p>{{ form.activities.label }} {{ form.activities }} {{ form.activities.description }}</p>
+        {% endif %}
+        {% if form.notes %}
```
This seems like an unnecessary check to make

aaroncohen73 commented on app/templates/layout.html:
```diff
@@ -2,23 +2,84 @@
-        {% endwith %}
+        {# Sidebar navigation #}
+        <nav>
+            <ul id="slide-out" class="side-nav fixed">
```
The sidenav bar is missing links to create/view projects

aaroncohen73 commented on app/__init__.py:
```diff
@@ -40,9 +41,23 @@
     print '         default. This is insecure, please generate a new one'
 
+assets = Environment(app)
+indexjs = Bundle('../js/index.js', filters='rjsmin', output='js/index.js')
```
When this is run after a `docker-compose up`, the files are owned by root. It's not like we'll need to edit these files by hand, but they should be renamed to `*.min.js` and/or put into a separate directory.

aaroncohen73 commented on app/templates/view_users.html:
```diff
@@ -81,3 +62,20 @@
+$(document).ready(function() {
+    var tableSelector = "#view-users table";
+    var perPage = 10;
+    var view_user_times_root = "{{ url_for('view_user_times') }}";
```
`view_user_times` doesn't work

aaroncohen73 reviewed (changes requested):
Overall, the website design looks pretty good. However, there are some things that aren't working. In addition to some of the notes I've left in the code,
- Table sorting doesn't work at all
- Multiselect inputs on forms appear as if an option is selected even if no option has been selected
- Form clearing doesn't work
- The pagination controls have disappeared entirely for some reason (There's an error in the browser console that might be helpful in figuring that one out).

subnomo committed:
Rename minified scripts, fix them being owned by root

subnomo committed:
Remove unnecessary form check

subnomo committed:
Add projects to sidenav

subnomo committed:
Add is_admin check to login page

subnomo committed:
Fix pagination

Will style soon

subnomo committed:
Add disabled default for multiselect fields

TODO: Some of the edit_* pages are broken, and all of the *_projects pages are broken

aaroncohen73 commented:
There seem to be just a few more things that need fixing before this is ready to merge. Fine-tuning the website design can come later.
- On some pages, not all of the sidebar links appear. For example, on the "Edit time" page the sidebar links to creating activities and projects don't appear. On some pages (The create/view projects pages and all of the edit and delete pages), the sidebar just shows the link to the login page.
- The edit project and view_user_times pages are broken.
- The following form inputs appear to be broken: The members/managers/spectators selection boxes on the "Create Project" page, all of the checkbox inputs

subnomo committed:
Post-rebase cleanup

subnomo committed:
Remove unnecessary filter

subnomo committed:
Remove duplicate scripts

subnomo committed:
Improve select fields

subnomo committed:
Fix *_project pages

subnomo committed:
Fix edit_* and delete_* pages

subnomo committed:
flake8

subnomo committed:
Add sidebar to view_user_times

subnomo commented:
@aaroncohen73 Thanks for the feedback, everything *should* be fixed now. Of course there are some usability tweaks to make, but that comes later.

Kennric commented:
This looks pretty good to me - not perfect, but I'd rather get it in in and fix it than sit on it waiting for perfection.
